src/Native/System.Security.Cryptography.Native/pal_asn1.cpp

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "pal_asn1.h"
+
+#include <openssl/objects.h>
+
+static_assert(NID_undef == 0, "Managed code relies on NID_undef == 0");
+static_assert(NID_secp224r1 == 713, "Managed code relies on NID_secp224r1 == 713");
```

Comment:
I would put PAL_NID_* constants in the .h. The idea should be that by reading the .h, you get everything you need to write the interop in C#. And then assert vs. the PAL_* constant.